### ISSUE TITLE
What's the point of `root`?

### ISSUE THREAD

**Issue by pascal-mueller** [air-verse/air#603]
I have this basic config file. I changed the root dir to `cmd`, not sure if that's good practice to begin with but then I figured out, I also have to manually adapt the build command and att `cmd` there as well. I assumed `root` would influence that, no?

Is that done correctly? It's hard to tell for me because the README doesn't really talk about it.

```
root = "./cmd"
testdata_dir = "testdata"
tmp_dir = "tmp"

[build]
  args_bin = []
  bin = "./tmp/main"
  cmd = "go build -o ./tmp/main ./cmd"
  delay = 1000
  exclude_dir = ["assets", "tmp", "vendor", "testdata"]
  exclude_file = []
  exclude_regex = ["_test.go"]
  exclude_unchanged = false
  follow_symlink = false
  full_bin = ""
  include_dir = []
  include_ext = ["go", "tpl", "tmpl", "html"]
  include_file = []
  kill_delay = "0s"
  log = "build-errors.log"
  poll = false
  poll_interval = 0
  post_cmd = []
  pre_cmd = []
  rerun = false
  rerun_delay = 500
  send_interrupt = false
  stop_on_error = false

[color]
  app = ""
  build = "yellow"
  main = "magenta"
  runner = "green"
  watcher = "cyan"

[log]
  main_only = false
  time = false

[misc]
  clean_on_exit = false

[proxy]
  app_port = 0
  enabled = false
  proxy_port = 0

[screen]
  clear_on_rebuild = false
  keep_scroll = true

```

**Comment by xiantang** [comment]
we use root to find relative path of tmp dir or bin path. in your case u should set root as `.`

**Comment by dw-0** [comment]
I was wondering the same while configuring air for a monorepo approach with nx was kinda confusing with a lot of trial and error involved. I thought the path specified for `root` will be used for everything that is configured further down the line. But that is not the case. The availability of a "root" path lets assume, that everything else in the config has to be configured _relative_ to that root path. But configuring root as `root = apps/backend` still requires bin and cmd to be configured as follows:
```toml
bin = "apps/backend/tmp/main.exe"
cmd = "go build -o apps/backend/tmp/main.exe apps/backend/cmd/main.go"
```

Also, it seems that root is used for the filewatcher:
https://github.com/air-verse/air/blob/326e87d6fbe92007dd65affd89616f19806eb0f4/runner/engine.go#L100

In conclusion: root is used for the filewatcher, and the generation of the tmp path. It is not used for the bin path (although i could spot some code that lets assume it is doing so) and the cmd.

**Comment by ankon** [comment]
> In conclusion: root is used for the filewatcher

... and this has an interesting implication: `include_dir` is relative to the root, _even if the path given given in `include_dir` is absolute_. If you give an absolute path, it simply never matches.

**Comment by xiantang** [comment]
point of root:
1. The working dir, air will be on root, to monitor the files under root with relative paths eg:include_dir exclude_dir .
2. for `bin` also base on your root

The wrong thing in your config is 
     root = "./cmd"
     tmp_dir = "tmp"

     [build]
       bin = "./tmp/main"
       cmd = "go build -o ./tmp/main ./cmd"  # it will seach ./cmd/cmd , because its already under ./cmd now
